src/xpublisher.hpp

```diff
@@ -29,13 +31,21 @@ namespace xeus
 
         std::string get_port() const;
 
+        template <class C>
+        inline void set_iopub_cb(zmq::multipart_t(C::*cb)(const std::string&), C* data)
```

Comment:
`xpublisher` could store an `std::function` bounded to `xserver_zmq/xserver_zmq_split::serialize_iopub_message`, that would be initialized by the server and passed to the constructor of the `xpublisher`.